### PR TITLE
add linters for no duplicates and no multiline

### PR DESCRIPTION
**Pull request by lsylvester**
Adds linters to ensure that there are no multiline names and no duplicate names.

### CODE REVIEW

File: lib/linters/duplicate-names.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+      counts[sample.name] = 0;
+    }
+    counts[sample.name] = counts[sample.name] + 1;
+  })
+  var errors = []
+  _.forEach(counts, function(count, name){
+    if(count > 1){
+      errors = errors.concat(name + ' (' + count + ' instances)')
+    }
+  })
```
Comment (rprieto) [inline lib/linters/duplicate-names.js]: What about:

```js
var counts = _.countBy(names);
var dupes = _.pick(counts, function(value) { return value > 1; });
var errors = _.map(dupes, function(val, key) {
  return key + ' (' + val + ' instances)';
});
```

File: lib/render.js [hunk 2]
```diff
@@ -9,12 +9,38 @@ var options  = require('./options');
...
+  var errors = instance.lint(model, options).errors
+
+  if(errors.length){
+    switch(options){
+      case 1:
+        console.warn('WARNING: ' + name + " found invalid samples: " + errors.join(", "))
+        break;
+      case 2:
+        throw(new Error(name + " found invalid samples: " + errors.join(", ")));
+        break;
```
Comment (rprieto) [inline lib/render.js]: Minor comment, could we `console.error` here, and then do a `process.exit(1)` at the end if necessary, so all errors get printed at once?

**Comment by rprieto** [comment]
A few comments but otherwise looks great :+1: Very nice one!

**Comment by lsylvester** [comment]
@rprieto Thanks for your feedback. I have updated based on your feedback

File: lib/render.js [hunk 3]
```diff
@@ -9,12 +9,41 @@ var options  = require('./options');
 
 exports.toDisk = function(model) {
   var opts = options.get();
+  _.forEach(opts.lint, function(options, name){
+    lint(name, options, model);
+  })
```
Comment (rprieto) [inline lib/render.js]: I'm not sure I understand the change. I meant something like:

```
```

Comment (lsylvester) [inline lib/render.js]: Sorry - didn't understand what you meant before. I have updated it now.

**Comment by rprieto** [comment]
Thanks, great feature to have :+1:

**Pull request merged**